packages/@ngtools/webpack/src/plugin.ts

```diff
@@ -461,8 +461,8 @@ export class AotPlugin implements Tapable {
             if (this.skipCodeGeneration) {
               this._lazyRoutes[k] = lazyRoute;
             } else {
-              const lr = path.relative(this.basePath, lazyRoute.replace(/\.ts$/, '.ngfactory.ts'));
-              this._lazyRoutes[k + '.ngfactory'] = path.join(this.genDir, lr);
+              const lr = path.relative(this.basePath, lazyRoute.replace(/(\.d)?\.ts$/, '.ngfactory.ts'));
+              this._lazyRoutes[k + '.ngfactory'] = path.join(this.genDir, path.join('/', lr));
```

Comment:
And the purpose of the join is to ensure lr cannot traverse up the directory tree and escape the $$_gendir directory. Assuming the following:

- basePath: '/project/src'
- genDir: '/project/src/$$_gendir
- lazyRoute: '/project/node_modules/package/lazy.module.ngfactory.ts'.

lr would then be '../node_modules/package/lazy.module.ngfactory.ts' and then the resulting _lazyRoutes path would end up being '/project/src/node_modules/package/lazy.module.ngfactory.ts', but the actual AOT factory file would be in '/project/src/$$_gendir/node_modules/package/lazy.module.ngfactory.ts'